superset/db_engine_specs/presto.py

```diff
@@ -373,6 +377,55 @@ def select_star(
             presto_cols,
         )
 
+    @classmethod
+    def estimate_statement_cost(cls, statement, database, cursor, user_name):
+        db_engine_spec = database.db_engine_spec
+        parsed_query = ParsedQuery(statement)
+        sql = parsed_query.stripped()
+
+        SQL_QUERY_MUTATOR = config.get("SQL_QUERY_MUTATOR")
+        if SQL_QUERY_MUTATOR:
+            sql = SQL_QUERY_MUTATOR(sql, user_name, security_manager, database)
+
+        sql = f"EXPLAIN (TYPE IO, FORMAT JSON) {sql}"
+
+        db_engine_spec.execute(cursor, sql)
+        polled = cursor.poll()
+        while polled:
+            time.sleep(0.2)
+            polled = cursor.poll()
+        data = db_engine_spec.fetch_data(cursor, 1)
+
+        first = data[0][0]
+        result = json.loads(first)
+        estimate = result["estimate"]
+
+        def humanize(value, suffix):
+            if value == "NaN":
+                return value
+
+            prefix = ""
+            symbols = ["K", "M", "G", "T", "P", "E", "Z", "Y"]
+            while value > 1000 and symbols:
+                prefix = symbols.pop(0)
+                value //= 1000
+
+            return f"{value} {prefix}{suffix}"
+
+        cost = {}
+        columns = [
```

Comment:
What happens if the presto format changes? Will we need to make changes here? I know presto likes to add a bunch of breaking changes in new versions...

Reply:
Changes won't break this, but might render the response empty. Not sure how to best deal with this.

Reply:
I guess we'll just need to keep an eye on it

Reply:
This is a feature we're expecting to use a lot, so we'll have a good number of eyes. But ideally we'd have integration tests with Presto.